### ISSUE TITLE
Feature request: Shares based on tag

### ISSUE THREAD

**Issue by jasonbayton** [nextcloud/server#1291]
Context: https://help.nextcloud.com/t/share-content-by-tag/3011

> I am a photographer and have a big database of photos in my cloud. Sometimes a (prospective) client asks for photos that meet criteria x, y and z. It would be nice if I could create a share for exactly the photos that I tagged with these criteria. Folders are too clumsy for that.

With the introduction and wider focus on tags and workflows, this would be a pretty nice function to have. Perhaps tagged files could reside within "virtual" folders that could benefit from the same functionality as actual folders (shares, public upload, etc) though with the added benefit of being entirely dynamic; as files have tags applied and removed, so too does the membership to these virtual folders change automatically.

**Comment by mbiebl** [comment]
I created a related issue for the gallery app: https://github.com/nextcloud/gallery/issues/277

**Comment by ftoral** [comment]
:+1: I vote for that.

I search for years an [anything] manager (mainly a photo album manager) that could share by tags and not only by folders/directories, and I do not find something fitting that simple requirement.

I think I could participate to a bounty for that feature !

**Comment by jasonbayton** [comment]
Yes it'd be great. Gallery should definitely get it too so totally fine it was created there, but I'm mainly interested in doing this with files.

**Comment by benklop** [comment]
are there currently blockers to implementing this feature? I'd love to see this happen, and am a developer, though I don't have much familiarity with the nextcloud codebase.

**Comment by basti1982** [comment]
Is there any news regarding this feature?

**Comment by basti1982** [comment]
I almost have the same situation, I want to create albums that I can easily share with other people. I really like the Google Photos functionality: select files by checkbox, hit + button and select create album. A virtual Album is created with the already existing files, that can be easily shared. 
I think that this functionality should be applied to the NC file manager and should not be restricted to Gallery only. It would even be and advantage in comparison to GDrive.

Creating shared folders/albums by using file copy is just a big waste of storage space, imho.

**Comment by wiwie** [comment]
I share some of my photos with my parents, but don't want to share all of them.

So currently I have to copy over all photos into a separate folder and share it with them.

I was looking into piwigo, as it can share and manage photos in virtual folders based on tags. But the Android client is basically non existent and hence it's a pain to upload new pictures...

Would be awesome to have this feature implemented in nextcloud!

**Comment by Spartachetto** [comment]
I am wondering if a solution to this issue could be a way to transform a tag in a "collection" ( see #11015 ).

Two further improvements would be
- transform a combination of tags in a collection (e.g.: "family" and not "private")
- include the files with a specific tag in a collection routinarily. This could be performed maybe by an automated workflow

**Comment by bananajota** [comment]
Hello! I'm interested on this feature too.  Is something new or some update about this? Thank you very much,

**Comment by aleksandrmetik** [comment]
Very useful feature. Could you please implement this? Thanks

**Comment by Spartachetto** [comment]
This will be dealt by the workflow backend overhaul (see #16682) and the other workflow improvements. Great news!!!

**Comment by jospoortvliet** [comment]
Don't consider it ready, please, the fact that some people work on it doesn't mean it is done next week ;-)

If you want this, the best thing continues to be: get involved and work on it, or contact Nextcloud GmbH and get a project to implement what you want/need.

**Comment by saspol** [comment]
Any progress with 'share by tag' feature? Im also interested.

**Comment by saspol** [comment]
Any progress with 'share by tag' feature? Im also interested.
Now not possible even select all files with same tag and copy it to separate folder to share. That's terrible lack of functionality.

**Comment by FALErwi** [comment]
Any progress with 'share by tag' feature?
It would be great feature 💐😔

**Comment by ACiDGRiM** [comment]
This feature has a lot of value to created dynamic search folders so users don't have to copy data into a public folder, just tagging it will add to an existing share.

**Comment by aleksandrmetik** [comment]
Guys, is there any progress here? Thanks

**Comment by ACiDGRiM** [comment]
I think this could be "easily" achieved by using the similar functionality of sharing files between NC users where the folder "appears" in their nextcloud files, but isn't actually an item in that user's Filesystem folder.

Once this reference folder is created, the contents can be listed by quering SELECT "paths" from Tags where Tag is "cats" when the folder is opened by either a public share or an internal NC user.

I don't think this needs a major new component built into NC

**Comment by CodeRoutineDev** [comment]
This feature would be such an awesome thing to have.

**Comment by JcMinarro** [comment]
Any update on it?
It would be really useful

**Comment by lorax42** [comment]
I'm just here to say, that there's still a lot of interest in this Issue.
I know that it's a big feature, but I just don't want it to be forgotten ^^
